admin/add.c

```diff
@@ -186,8 +186,8 @@ read_file(FILE *f)
 
             if ((tmp = realloc(res, alloced + (alloced > 1))) == NULL)
                 err(1, "Out of memory");
+            p = tmp + alloced;
```

Comment:
This really doesn't look right.  Shouldn't it be `p = tmp + len`?

Reply:
Yes, it would be much clearer, as we can clearly see len incrementing on line 195